### PR TITLE
feat: CloudWatch SDK

### PR DESCRIPTION
**Pull request by andrezimmermann**
I've added the CloudWatch SDK, based of the SSM SDK scaffolding. 

Over the documentation I just pretended that there is the cloudwatch quickstart and coded an HelloWorld example.

Work for #1150

---
On another note, I realized while coding the example that I actually wanted the CloudWatchLogs SDK, will be working on it if this PR is passable

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -237,6 +237,23 @@
                 <artifactId>quarkus-amazon-secretsmanager-deployment</artifactId>
                 <version>${project.version}</version>
             </dependency>
+
```
Comment (scrocquesel) [inline bom/pom.xml:240]:
```suggestion
```

File: bom/pom.xml [hunk 2]
```diff
@@ -237,6 +237,23 @@
...
+                <groupId>io.quarkiverse.amazonservices</groupId>
+                <artifactId>quarkus-amazon-cloudwatch</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.quarkiverse.amazonservices</groupId>
+                <artifactId>quarkus-amazon-cloudwatch-deployment</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
```
Comment (scrocquesel) [inline bom/pom.xml:256]:
```suggestion
```

File: cloudwatch/runtime/pom.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    <parent>
+        <groupId>io.quarkiverse.amazonservices</groupId>
+        <artifactId>quarkus-amazon-cloudwatch-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>quarkus-amazon-cloudwatch</artifactId>
+    <name>Quarkus - Amazon Services - CloudWatch - Runtime</name>
+    <description>Connect to Amazon Secrets Manager</description>
```
Comment (scrocquesel) [inline cloudwatch/runtime/pom.xml:13]: change the description

File: pom.xml [hunk 4]
```diff
@@ -16,6 +16,7 @@
     <module>build-parent</module>
     <module>crt</module>
     <module>common</module>
+    <module>cloudwatch</module>
     <module>devservices</module>
```
Comment (scrocquesel) [inline pom.xml:20]:
```suggestion
    <module>devservices</module>
    <module>cloudwatch</module>
```

**Review by scrocquesel — CHANGES_REQUESTED** [review]
Awesome. Just a few remarks and you missed the reference from the docs/pom.xml (see https://github.com/quarkiverse/quarkus-amazon-services/blob/65f5cffa9908a4bffb9d4fb5f179fa81cc3a6dd4/docs/pom.xml#L129

pages in the included directory must be copied automatically by maven at the end of the build

**Comment by andrezimmermann** [comment]
Thanks for the review, I will adjust these tomorrow and do the same for #1154

**Comment by scrocquesel** [comment]
Could you add a small integration test based on the quickstart to assert it works with LocalStack.

**Comment by andrezimmermann** [comment]
Sure, I let me do something by Monday

File: cloudwatch/deployment/src/test/java/io/quarkus/amazon/cloudwatch/deployment/CloudWatchAsyncClientTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Inject
+    CloudWatchAsyncClient client;
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
+
+    @Test
+    public void shouldBeAbleToPutMetricData() {
```
Comment (scrocquesel) [inline cloudwatch/deployment/src/test/java/io/quarkus/amazon/cloudwatch/deployment/CloudWatchAsyncClientTest.java:32]: I would prefer to have tests in the integration-tests project. It is the only place where we tests all combination of client types and native build.

Comment (andrezimmermann) [reply cloudwatch/deployment/src/test/java/io/quarkus/amazon/cloudwatch/deployment/CloudWatchAsyncClientTest.java]: Right, I wasn't aware, I've delete these tests and move into the integration module, trying to mimic the other tests

**Review by scrocquesel — APPROVED** [review]

**Comment by scrocquesel** [comment]
@all-contributors please add @andrezimmermann for code

**Pull request merged**